src/os/bluestore/BlueStore.cc

```diff
@@ -5517,6 +5582,149 @@ int BlueStore::_read_bdev_label(CephContext* cct, const string &path,
   return 0;
 }
 
+/**
+  Reads device label.
+  cct             - CephContext, as usual
+  path            - Path to block device, as conf.bluestore_block_path defines.
+  *out_label      - Filled if reading of label is considered successful.
+  *out_valid_positions - List of locations that contained valid labels.
+  *out_is_multi  -  Whether the label is regular or multi label with epoch.
+  *out_epoch     -  Epoch of label.
+
+  Returns:
+  0    When all label are read
+  1    When some, but not all labels are read
+  -ENOENT Otherwise
+
+*/
+int BlueStore::_read_main_bdev_label(
+  CephContext* cct,
+  const string &path,
+  bluestore_bdev_label_t *out_label,
+  std::vector<uint64_t>* out_valid_positions,
+  bool* out_is_multi,
+  int64_t* out_epoch)
+{
+  dout(10) << __func__ << dendl;
+  ceph_assert(out_label);
+  // go and try read all possible bdev labels.
+  // if only first bdev label is correct, it must not have "multi=yes" key.
+  int64_t epoch = -1;
+  bool all_labels_valid = true;
+  for (uint64_t position : bdev_label_positions) {
+    bluestore_bdev_label_t label;
+    int r = _read_bdev_label(cct, path, &label, position);
+    if (r == 0) {
+      auto i = label.meta.find("multi");
+      bool is_multi = i != label.meta.end() && i->second == "yes";
+      if (position == BDEV_LABEL_POSITION && !is_multi) {
```

Comment:
```suggestion
      if (position == BDEV_FIRST_LABEL_POSITION && !is_multi) {
```
less ambigous this way imo

Reply:
Con:
1) The problem is that for block-main it is the first label position, but for block-db and block-wal it is the only label.
Pro: 
1) Granted, only label is first label.
2) If we make multiple labels for block-db and block-wal, new name would fit.

Do we change?

Reply:
Renamed to BDEV_FIRST_LABEL_POSITION